kapitan/refs/secrets/vaultkv.py

```diff
@@ -0,0 +1,295 @@
+                kwargs.update(_kwargs)
+                return cls(obj['data'], from_base64=True, **kwargs)
+
+        except IOError as ex:
+            if ex.errno == errno.ENOENT:
+                logger.debug("no such file or directory {}".format(kwargs['inventory_path']))
+                return None
+
+        except KeyError:
+            raise VaultError("Could not fetch VaultSecret: vaultkv parameters missing")
+
+    def reveal(self):
+        """
+        Returns decrypted data
+        """
+        # can't use super().reveal() as we want bytes
+        try:
+            if self.encoding == 'base64':
+                self.data = base64.b64decode(self.data, validate=True)
+
+        except b_error:
+            exit(
+                "non-alphabet characters in the data"
+            )
+
+        return self._decrypt()
+
+    def _decrypt(self):
+        """
+        Authenticate with Vault server & returns value of the key from secret
+
+        :returns: secret in plain text
+        """
+        try:
+            client = vault_obj(self.parameter)
+            # token will comprise of two parts path_in_vault:key
+            data = self.data.decode('utf-8').split(':')
+            return_data = ''
+            if self.parameter.get('engine') == 'kv':
+                response = client.secrets.kv.v1.read_secret(path=data[0],
+                                                            mount_point=self.parameter.get('mount','secret'))
+                return_data = response['data'][data[1]]
+            else:
+                response = client.secrets.kv.v2.read_secret_version(path=data[0],
+                                                                    mount_point=self.parameter.get('mount','secret'))
+                return_data = response['data']['data'][data[1]]
+            client.adapter.close()
+        except Forbidden:
+            VaultError(
+                'Permission Denied. '+
+                'make sure the token is authorised to access {path} on Vault'.format( path=data[0])
+            )
+        except InvalidPath:
+            VaultError(
+                '{path} does not exist on Vault secret'.format(path=data[0])
+            )
+
+        if return_data is '':
+            VaultError(
+                "'{key}' doesn't exist on '{path}'".format(key=data[1], path=data[0])
+            )
+        return return_data
+
+    def dump(self):
+        """
+        Returns dict with keys/values to be serialised.
+        """
+        return {"data": self.data, "encoding": self.encoding,
+                "type": self.type_name}
+
+class VaultBackend(Base64RefBackend):
+    def __init__(self, path, target_name, inventory_path, ref_type=VaultSecret):
+        "init VaultBackend ref backend type"
+        super().__init__(path, ref_type)
+        self.type_name = 'vaultkv'
+        self.ref_type = ref_type
+        self.target_name = target_name
+        self.inventory_path = inventory_path
```

Comment:
Where is this inventory_path actually used? Do we need to set it here and in `RefController` or is it a leftover after changes?

Reply:
`inventory_path` will be used when we are revealing the secrets, this was actually useful only for testing purpose till now.